localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -7,34 +7,238 @@
+    reached.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.queues = []
+
+    def add_queue(self, queue: SqsQueue):
+        self.queues.append(queue)
+
+    def remove_queue(self, queue: SqsQueue):
+        self.queues.remove(queue)
+
+    def iter_queues(self) -> Iterable[SqsQueue]:
+        return iter(self.queues)
+
+
+class QueueManager:
+    queues: dict[str, StandardQueue]
+    queue_lock: threading.RLock
+    queue_update_worker: EventQueueUpdateWorker
+
+    def __init__(self):
+        self.queues = {}
+        self.queue_lock = threading.RLock()
+        self.queue_update_worker = EventQueueUpdateWorker()
+        self.queue_update_worker.start()
+
+    def get_queue(self, queue_name: str):
+        if queue_name not in self.queues:
+            raise ValueError("Queue not available")
+        return self.queues[queue_name]
+
+    def create_queue(self, queue_name: str) -> SqsQueue:
+        """
+        Creates a queue.
+        :param queue_name: Queue name, has to be unique
+        :return: Queue Object
+        """
+        with self.queue_lock:
+            if queue_name in self.queues:
+                return self.queues[queue_name]
+
+            # Truncate function name to ensure queue name limit of max 80 characters
+            queue = StandardQueue(
+                name=queue_name,
+                region="us-east-1",
+                account_id=config.INTERNAL_RESOURCE_ACCOUNT,
+            )
+            self.queues[queue_name] = queue
+            self.queue_update_worker.add_queue(queue)
+        return queue
+
+    def delete_queue(self, queue_name: str) -> None:
+        with self.queue_lock:
+            if queue_name not in self.queues:
+                raise ValueError("Queue not available")
+
+            queue = self.queues.pop(queue_name)
+            self.queue_update_worker.remove_queue(queue)
+
+
+class FakeSqsClient:
+    def __init__(self, queue_manager: QueueManager):
+        self.queue_manager = queue_manager
+
+    def create_queue(
+        self, QueueName: String, attributes: QueueAttributeMap = None, tags: TagMap = None
+    ) -> CreateQueueResult:
+        self.queue_manager.create_queue(queue_name=QueueName)
+        return {"QueueUrl": QueueName}
+
+    def delete_queue(self, QueueUrl: String) -> None:
+        self.queue_manager.delete_queue(queue_name=QueueUrl)
+
+    def get_queue_attributes(
+        self, QueueUrl: String, AttributeNames: AttributeNameList = None
+    ) -> GetQueueAttributesResult:
+        queue = self.queue_manager.get_queue(queue_name=QueueUrl)
+        result = queue.get_queue_attributes(AttributeNames)
+        return {"Attributes": result}
+
+    def purge_queue(self, QueueUrl: String) -> None:
+        queue = self.queue_manager.get_queue(queue_name=QueueUrl)
+        queue.clear()
+
+    def receive_message(
+        self,
+        QueueUrl: String,
+        attribute_names: AttributeNameList = None,
```

Comment:
Is the mixed naming style of CamelCase and snake_case intentional here?

Reply:
No, should all be camel case for API compatibility, good catch! Missed one